### PR TITLE
Feature: Implementation of RFC Attributes

### PR DESCRIPTION
**Pull request by chris-rock**
This PR implements parts of RFC Attributes #719

To use it:
`$ inspec exec examples/profile-attribute --attrs examples/profile-attribute.yml`

An attributes file looks like:

```
cat examples/profile-attribute.yml 
user: bob
password: secret
```

The usage within a profile is as following:

```
val_user = attribute('user', required: true)

describe val_user do
  it { should eq 'johndoe' }
end
```

### CODE REVIEW

File: examples/profile-attribute/controls/example.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# encoding: utf-8
+val_user = attribute('user', default: 'alice', description: 'An identification for the user')
+val_password = attribute('password', description: 'A value for the password')
+
+# that works
+describe 'bob' do
+  it { should eq val_user }
```
Comment (arlimus) [inline examples/profile-attribute/controls/example.rb]: Inverting feels more readable:

```
describe val_user do
  it { should eq 'bob' }
```

File: examples/profile-attribute/controls/example.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+end
+
+describe 'secret' do
+  it { should eq val_password }
```
Comment (arlimus) [inline examples/profile-attribute/controls/example.rb]: ditto

File: examples/profile-attribute/controls/example.rb (hunk 1 above)
Comment (arlimus) [inline examples/profile-attribute/controls/example.rb]: ditto vv

**Comment by arlimus** [comment]
Awesome work @chris-rock !!

Adding the ncessary tests should round this up.

File: examples/profile-attribute/controls/example.rb (hunk 1 above)
Comment (chris-rock) [inline examples/profile-attribute/controls/example.rb]: Yeah, I like that...

**Pull request merged**